application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    private Result<PendingRemindersRecord> getPendingReminders(Guild guild, User user) {
+        return database.read(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.GUILD_ID.eq(guild.getIdLong())
+                .and(PENDING_REMINDERS.AUTHOR_ID.eq(user.getIdLong())))
+            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
+            .fetch());
+    }
+
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = (int) Math.ceil(pendingReminders.size() / (double) MAX_PAGE_LENGTH);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
+
+        if (pendingReminders.isEmpty()) {
+            remindersEmbed.setDescription("No pending reminders");
+        } else {
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: " + pageToShow + "/" + totalPages);
+
+                getRemindersPage(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+            }
+        }
+
+        return listMessage.addEmbeds(remindersEmbed.build()).build();
+    }
+
+    private List<Button> createPageTurnButtons(int pageNumber, int totalPages) {
+        int previousButtonTurnPageBy = -1;
+        Button previousButton =
+                createPageTurnButton(PREVIOUS_BUTTON_LABEL, pageNumber, previousButtonTurnPageBy);
+        if (pageNumber <= 1) {
+            previousButton = previousButton.asDisabled();
+        }
+
+        int nextButtonTurnPageBy = 1;
+        Button nextButton =
+                createPageTurnButton(NEXT_BUTTON_LABEL, pageNumber, nextButtonTurnPageBy);
+        if (pageNumber >= totalPages) {
+            nextButton = nextButton.asDisabled();
+        }
+
+        return List.of(previousButton, nextButton);
+    }
+
+    private Button createPageTurnButton(String label, long pageNumber, int turnPageBy) {
+        return Button.primary(
+                generateComponentId(String.valueOf(pageNumber), String.valueOf(turnPageBy)), label);
+    }
+
+    private static List<PendingRemindersRecord> getRemindersPage(
+            Result<PendingRemindersRecord> remindersRecords, int pageNumber) {
+        List<PendingRemindersRecord> reminders = new ArrayList<>(MAX_PAGE_LENGTH);
+
+        int start = (pageNumber - 1) * MAX_PAGE_LENGTH;
+        int end = Math.min(start + MAX_PAGE_LENGTH, remindersRecords.size());
+
+        for (int i = start; i < end; i++) {
+            reminders.add(remindersRecords.get(i));
+        }
+
```

Comment:
You can use `remindersRecords.subList(start, end)`

Reply:
i tried looking for a method which does that but i gave up too quickly :joy: